### PR TITLE
Transform v1 -> v2 run.resources.rules

### PR DESCRIPTION
**Pull request by EasyRhin0**

### CODE REVIEW

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 1]
```diff
@@ -222,9 +226,8 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
             files.Should().NotBeNull();
             files.Count.Should().Be(3);
 
-            FileData fileData = null;
+            FileData fileData = files["file:///home/list.txt"];
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: Nice cleanups.

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 2]
```diff
@@ -233,8 +236,10 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
             fileData.Hashes.Should().NotBeNull();
             fileData.Hashes[0].Algorithm.Should().Be("sha-256");
             fileData.Hashes[0].Value.Should().Be("d7a8fbb307d7809469ca9abcb0082e4f8d5651e46d3cdb762d02d0bf37c9e592");
+            fileData.Properties.Should().BeNull();
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: Affirm.

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 3]
```diff
@@ -254,6 +260,107 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
...
+                    {
+                      ""rules"": {
+                        ""C2001"": {
+                          ""id"": ""C2001"",
+                          ""shortDescription"": ""A variable was used without being initialized."",
+                          ""messageFormats"": {
+                            ""default"": ""Variable \""{0}\"" was used without being initialized.""
+                          },
+                          ""properties"": {
+                            ""some_key"": ""FoxForceFive""
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: LOL

Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifTransformerTests.cs]: :)

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 4]
```diff
@@ -254,6 +260,107 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
...
+                      ]
+                    }
+                  ]
+                }";
+
+            SarifLog v2Log = TransformVersionOneToCurrent(v1LogText);
+
+            v2Log.Runs.Should().NotBeNull();
+            v2Log.Runs.Count.Should().Be(1);
+            v2Log.Runs[0].Resources.Should().NotBeNull();
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: Let's use our shiny new `run` temp var here too.

Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifTransformerTests.cs]: Ah missed that one

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 5]
```diff
@@ -254,6 +260,107 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
...
+            rule.Properties.Should().NotBeNull();
+            rule.Properties["some_key"].SerializedValue.Should().Be("\"FoxForceFive\"");
+
+            rule = v2Log.Runs[0].Resources.Rules["C2002"];
+
+            rule.Id.Should().Be("C2002");
+            rule.Name.Should().BeNull();
+            rule.ShortDescription.Should().BeNull();
+            rule.FullDescription.Should().NotBeNull();
+            rule.FullDescription.Text.Should().Be("Catfish season continuous hen lamb include dose copy grant.");
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: Nice catching all the properties this way.

File: src/Sarif.UnitTests/SarifTransformerTests.cs (hunk 4 above)
Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifTransformerTests.cs]: Done

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 6]
```diff
@@ -254,6 +260,107 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
...
+            rule.Id.Should().Be("C2003");
+            rule.Name.Should().NotBeNull();
+            rule.Name.Text.Should().Be("Rule C2003");
+            rule.ShortDescription.Should().NotBeNull();
+            rule.ShortDescription.Text.Should().Be("Rules were meant to be broken.");
+            rule.FullDescription.Should().NotBeNull();
+            rule.FullDescription.Text.Should().Be("Rent internal rebellion competence biography photograph.");
+            rule.Configuration.Should().NotBeNull();
+            rule.Configuration.Enabled.Should().Be(false);
+            rule.Configuration.DefaultLevel.Should().Be(RuleConfigurationDefaultLevel.Warning);
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: `"pass"` should map to `RuleConfigurationDefaultLevel.Note`.

Comment (EasyRhin0) [reply src/Sarif.UnitTests/SarifTransformerTests.cs]: Fixed

File: src/Sarif.UnitTests/SarifTransformerTests.cs [hunk 7]
```diff
@@ -254,6 +260,110 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
...
+            v2Log.Runs.Should().NotBeNull();
+            v2Log.Runs.Count.Should().Be(1);
+
+            Run run = v2Log.Runs[0];
+
+            run.Resources.Should().NotBeNull();
+            run.Resources.Rules.Should().NotBeNull();
+            run.Resources.Rules.Count.Should().Be(3);
+
+            Rule rule = v2Log.Runs[0].Resources.Rules["C2001"];
```
Comment (lgolding) [inline src/Sarif.UnitTests/SarifTransformerTests.cs]: Two more places for `run`, here and on Line 337.

**Pull request merged**